### DISCUSSION TITLE
Why can I not use the greater than operator with the default key word?

### DISCUSSION THREAD

**Discussion by TylerDM** [dotnet/csharplang#4477] [Q&A] [answered]
Today I tried to do the following as an experiment:
```csharp
public async Task<bool> CreateAsync(int id)
{
  if (id == default) throw new ArgumentDefaultException(nameof(id));
  if (id < default) throw new ArgumentOutOfRangeException(nameof(id));

  //...
}
```

But received the following error:
> Operator '<' cannot be applied to the operand 'default'

Why is this?  I assumed the compiler was replacing `default` with the equivalent literal value.  But if that was the case, this should have worked.  Maybe it's an oversight?

**Comment by 333fred** [comment, marked as answer]
The `default` literal keyword needs to be "target-typed" in order to figure out what type it should represent. We don't have any rules for inferring target-types for one side of an operator like `>` or `<` from the other side, so `id` cannot inform `default` what type it should be. I would also say that, IMO, this type of code relatively unclear, and it's much better to just use 0.
